### ISSUE TITLE
Sites that do not work

### ISSUE THREAD

**Issue by richd67** [hkgnp/logseq-recipeimporter-plugin#1]
Unfortunately all the sites I tried didnt work list below 
- https://www.gastrosenses.com/blog/beef-masala-curry/
- https://nishkitchen.com/indian-beef-curry/
- https://www.bbcgoodfood.com/recipes/beef-curry

**Comment by benjypng** [comment]
Hi! Are you on the latest version?

The first and last links actually worked for me, but not the second one.

**Comment by richd67** [comment]
Just tried the first one again having just checked that it is the most up to date version and am still getting this error 'There was an error parsing the url, Please file a github issue with the url'

**Comment by nicgentile** [comment]
Hi @hkgnp I can confirm what @richd67 said. No sites worked for me at all. Something is broken. Might is on [this build.](https://github.com/logseq/logseq/releases/download/nightly/Logseq-linux-x64-0.6.5+nightly.20220408.AppImage) on Debian.

Thanks.

**Comment by benjypng** [comment]
Thanks all. That is weird as it seems to be working on mine. 

Importing recipes is a bit of a hit and miss because of the way websites structure their recipe content. They do not all do it the same way resulting in a multitude of permutations!

**Comment by benjypng** [comment]
> Hi @hkgnp I can confirm what @richd67 said. No sites worked for me at all. Something is broken. Might is on [this build.](https://github.com/logseq/logseq/releases/download/nightly/Logseq-linux-x64-0.6.5+nightly.20220408.AppImage) on Debian.
> 
> Thanks.

Do you have an example to share?

**Comment by nicgentile** [comment]
https://www.kingarthurbaking.com/recipes/chocolate-wafers-recipe
https://www.seriouseats.com/choux-pastry
https://www.averiecooks.com/black-white-cookies-cream-cheese-chocolate-chip-dark-chocolate-dark-brown-sugar/
https://lovelyjubley.com/vegan-salted-dark-chocolate-chilli-pots/
https://www.allrecipes.com/gallery/chicken-foil-packet-recipes/

Regardless of the site, none of them work, and with the little that I know about these things, the error is instant, not delayed, so the request itself might not even be happening in the first place.

**Comment by benjypng** [comment]
> https://www.kingarthurbaking.com/recipes/chocolate-wafers-recipe https://www.seriouseats.com/choux-pastry https://www.averiecooks.com/black-white-cookies-cream-cheese-chocolate-chip-dark-chocolate-dark-brown-sugar/ https://lovelyjubley.com/vegan-salted-dark-chocolate-chilli-pots/ https://www.allrecipes.com/gallery/chicken-foil-packet-recipes/
> 
> Regardless of the site, none of them work, and with the little that I know about these things, the error is instant, not delayed, so the request itself might not even be happening in the first place.

Does the plugin even create a page after you paste the URL and press Enter or nothing happens?

**Comment by nicgentile** [comment]
Nothing happens. Literally, the immediate response is the error message, less than a microsecond after 'enter'. It's instantaneous, hence why I believe no request happens in the first place.

**Comment by benjypng** [comment]
> Nothing happens. Literally, the immediate response is the error message, less than a microsecond after 'enter'. It's instantaneous, hence why I believe no request happens in the first place.

Hmm, that it means something else isn't working as it is supposed to. I tried the links you provided above and some of them do work so not sure what is happening on your side. 

Anyway, I am in the process of trying to refactor the code to accommodate more recipe templates, although I must still caveat that it would not be 100% as the sites really have all sorts of permutations.

**Comment by pushakargaikwad** [comment]
https://shwetainthekitchen.com/creamy-vegetable-sandwich/
I think this site uses WordPress Recipe Maker Plugin.

**Comment by mstine** [comment]
Additional problem URLs:

- https://www.eatwell101.com/pepper-steak-stir-fry-recipe
- https://www.saltandlavender.com/italian-sausage-orzo/

Same error message got me to post these. Thanks!

**Comment by JCGitH** [comment]
Error received in Logsec states "Apologies, there was an error parsing the URL. Please file a Github issue with the URL."  The URL was https://www.allrecipes.com/recipe/230660/orange-honey-and-soy-chicken/   7/21/22

**Comment by briansunter** [comment]
Doesn't work for me either. https://thefoodcharlatan.com/pizza-dough-recipe/

**Comment by kensipe** [comment]
I'm getting the failures as well.   from dev tools console:

```
Access to XMLHttpRequest at 'https://www.seriouseats.com/choux-pastry' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
index.e2686684.js:1 Error: Network Error
    at a.register.e.exports (index.e2686684.js:1:213808)
    at b.onerror (index.e2686684.js:1:212385)
index.e2686684.js:1          GET https://www.seriouseats.com/choux-pastry 
```

a google search top hit is another plugin with the issue which @hkgnp seemed to have some awareness of:  https://github.com/hkgnp/logseq-mermaid-plugin/issues/4

**Comment by uhuang** [comment]
https://food52.com/recipes/62716-clear-wonton-broth
https://www.seriouseats.com/homemade-wonton-soup-recipe

Neither of the above work for me. I was, however, able to get the Beef Curry link in the original post to work.

**Comment by benjypng** [comment]
Yeah, it is hard to parse recipes as there is no real standard out there
that every website uses. I am still trying to find a better way to parse
them than the one in the plugin now.

Apologies.

**Comment by lukas-mertens** [comment]
@hkgnp 

https://cookidoo.de/recipes/recipe/de-DE/r282724

here is another one. To be honest: I would suggest adding a framework for scraping data from websites and letting the community contribute and maintain concrete implementations for certain websites. Keep the current "one-fits-all" algorithm and let users contribute custom ones, similar to how youtube-dl does it. Basically some abstract scraper class with concrete implementations where every implementation includes a regex at the beginning on which urls it should step in and do the scraping.

If you would do something like that, I can even imagine that we could write the scrapers almost automatically using ChatGPT/GPT3.0 by providing it with the abstract scraper class, an example implementation for one popular recipe site and the html of the page which it should scrape. That way we could contribute sites really fast and it would be scalable.

If one scraper works for multiple pages (e.g. because they are using the same CMS like wordpress, we would just have to extend the array of url-regexes).

I think this would greatly reduce the complexity of your code and would truly allow you to develop a one-fits-all solution.

The scraping part could even be a separate open source project to be honest, as I can believe, not only logseq users are interested in something like recipe-dl ^^ Basically youtube-dl but for recipes ;)

**Comment by lukas-mertens** [comment]
Actually I did some further research. It seems like this is a great blog-post:

https://www.benawad.com/scraping-recipe-websites/

Maybe that helps, especially the part about ld+json is what you could definitely implement.

**Comment by pablo03v** [comment]
https://www.gaumenfreundin.de/rote-linsen-lasagne/

**Comment by pablo03v** [comment]
https://biancazapatka.com/en/vegan-lasagna-with-lentils-and-zucchini/#recipe

**Comment by shotcollin** [comment]
None of these worked for me. I have plugin v1.0.2.
https://www.allrecipes.com/recipe/221321/sazerac-cocktail/
https://www.liquor.com/recipes/sazerac/
https://www.thespruceeats.com/sazerac-cocktail-recipe-760604

**Comment by danpalmer** [comment]
Latest version of the plugin, these don't work:

https://lucyandlentils.co.uk/recipe/courgette-lemon-orzo
https://www.recipetineats.com/vietnamese-pork-noodle-bowls/
https://www.recipetineats.com/golden-coconut-chicken-curry/#h-ingredients-in-golden-coconut-chicken-curry

**Comment by bpitrain** [comment]
V1.0.2 - this site doesn't work

[https://tasty.co/recipe/colombian-pan-de-bono](https://tasty.co/recipe/colombian-pan-de-bono)
